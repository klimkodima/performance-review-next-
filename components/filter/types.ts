export type FilterType = {
  name: string;
  value: boolean;
};
