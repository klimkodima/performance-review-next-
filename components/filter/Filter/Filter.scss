.filter {
  display: flex;
  justify-content: center;
  background-color: #f6f6f6;
  border: 2px solid #e3e3e3;
  min-height: 100%;
  width: 250px;
  flex-shrink: 0;

  &__wrapper {
    width: 100%;
    padding: 15px;
  }

  &__header {
    width: 100%;
    display: flex;
    justify-content: space-between;
    align-items: center;
    margin-bottom: 40px;
  }

  &__title {
    color: #818181;
    margin: 0;
    font-weight: 400;
    font-family: 'Roboto', 'Helvetica', 'Arial', sans-serif;
    font-size: 1rem;
    letter-spacing: 0.00938em;
  }

  .clear-button {
    color: #9f9f9f;
    padding: 20px;
    font-size: 1rem;
    line-height: 0;
  }
}
