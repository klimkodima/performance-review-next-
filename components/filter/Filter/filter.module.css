.filter {
  display: flex;
  justify-content: center;
  background-color: #f6f6f6;
  border: 2px solid #e3e3e3;
  min-height: 100%;
  width: 250px;
  flex-shrink: 0;
}

.filter__wrapper {
  width: 100%;
  padding: 15px;
}

.filter__header {
  width: 100%;
  display: flex;
  justify-content: space-between;
  align-items: center;
  margin-bottom: 40px;
}

.filter__title {
  color: #818181;
  margin: 0;
  font-weight: 400;
  font-family: 'Roboto', 'Helvetica', 'Arial', sans-serif;
  font-size: 1rem;
  letter-spacing: 0.00938em;
}

.clear__button {
  color: #9f9f9f;
  padding: 20px;
  font-size: 1rem;
  line-height: 0;
}