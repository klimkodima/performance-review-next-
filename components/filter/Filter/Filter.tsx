import { ReactElement, memo } from 'react';
import Button from '@mui/material/Button';
import { useTranslation } from 'next-i18next';
import { FilterList } from '../FilterList';

import { FilterType } from '../types';

import './Filter.scss';

type FilterPropsType = {
  teamFilter: FilterType[];
  levelFilter: FilterType[];
  auditorFilter: FilterType[];
};

const Filter = memo(
  ({
    levelFilter,
    teamFilter,
    auditorFilter
  }: FilterPropsType): ReactElement => {
    const { t } = useTranslation('common', {
      keyPrefix: 'filter'
    });
    return (
      <div className='filter'>
        <div className='filter__wrapper'>
          <div className='filter__header'>
            <h3 className='filter__title'>{t('FILTERS')}</h3>
            <Button className='clear-button'>{t('CLEAR ALL')}</Button>
          </div>
          <div>
            <FilterList title={t('TEAM')} filter={teamFilter} />
            <FilterList title={t('LEVEL')} filter={levelFilter} />
            <FilterList title={t('AUDITOR')} filter={auditorFilter} />
          </div>
        </div>
      </div>
    );
  }
);

export default Filter;

Filter.displayName = 'Filter';
