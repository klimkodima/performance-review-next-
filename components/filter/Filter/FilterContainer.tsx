import { ReactElement } from 'react';
import { useTranslation } from 'next-i18next';
import Filter from './Filter';
import { FilterType } from '../types';

export const mockTeamFilter: FilterType[] = [
  { name: 'Alpha', value: true },
  { name: 'Charlie', value: false },
  { name: 'X-Rays', value: false }
];

export const mockAuditorFilter: FilterType[] = [
  { name: 'Alex Vagner', value: true },
  { name: 'Tom Cruise', value: true }
];

const FilterContainer = (): ReactElement => {
  const { t } = useTranslation('filter');
  const mockLevelFilter: FilterType[] = [
    { name: t('Trainee'), value: true },
    { name: t('Junior'), value: true },
    { name: t('Middle'), value: false },
    { name: t('Senior'), value: false },
    { name: t('Team Lead'), value: false }
  ];
  return (
    <Filter
      teamFilter={mockTeamFilter}
      levelFilter={mockLevelFilter}
      auditorFilter={mockAuditorFilter}
    />
  );
};

export default FilterContainer;
