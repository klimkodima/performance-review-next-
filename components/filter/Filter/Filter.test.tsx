import renderer from 'react-test-renderer';
import Filter from './Filter';
import { mockAuditorFilter, mockTeamFilter } from './FilterContainer';
import { FilterType } from '../types';

const mockLevelFilter: FilterType[] = [
  { name: 'Trainee', value: true },
  { name: 'Junior', value: true },
  { name: 'Middle', value: false },
  { name: 'Senior', value: false },
  { name: 'Team Lead', value: false }
];

describe('Filter', () => {
  test('Matches snapshot', () => {
    const tree = renderer
      .create(
        <Filter
          teamFilter={mockTeamFilter}
          levelFilter={mockLevelFilter}
          auditorFilter={mockAuditorFilter}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
