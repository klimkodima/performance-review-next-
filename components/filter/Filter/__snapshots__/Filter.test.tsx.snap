// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter Matches snapshot 1`] = `
<div
  className="filter"
>
  <div
    className="filter__wrapper"
  >
    <div
      className="filter__header"
    >
      <h3
        className="filter__title"
      >
        FILTERS
      </h3>
      <button
        className="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root clear-button css-1e6y48t-MuiButtonBase-root-MuiButton-root"
        disabled={false}
        onBlur={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        CLEAR ALL
      </button>
    </div>
    <div>
      <div
        className="filter-list"
      >
        <ul
          className="MuiList-root MuiList-padding css-h4y409-MuiList-root"
          data-testid="filter-list-test"
        >
          <div
            className="filter-list__name"
          >
            <p
              className="MuiTypography-root MuiTypography-body1 title css-ahj2mt-MuiTypography-root"
            >
              TEAM
            </p>
            <button
              className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium plus-button css-78trlr-MuiButtonBase-root-MuiIconButton-root"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              +
            </button>
          </div>
          <li
            className="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
            disabled={false}
          >
            <div
              className="filter-list-item"
            >
              <p
                className="MuiTypography-root MuiTypography-body1 filter-list-item__filter-name css-ahj2mt-MuiTypography-root"
              >
                Alpha
              </p>
              <button
                className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium filter-list-item__delete-button css-78trlr-MuiButtonBase-root-MuiIconButton-root"
                disabled={false}
                onBlur={[Function]}
                onContextMenu={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={0}
                type="button"
              >
                +
              </button>
            </div>
          </li>
        </ul>
      </div>
      <div
        className="filter-list"
      >
        <ul
          className="MuiList-root MuiList-padding css-h4y409-MuiList-root"
          data-testid="filter-list-test"
        >
          <div
            className="filter-list__name"
          >
            <p
              className="MuiTypography-root MuiTypography-body1 title css-ahj2mt-MuiTypography-root"
            >
              LEVEL
            </p>
            <button
              className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium plus-button css-78trlr-MuiButtonBase-root-MuiIconButton-root"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              +
            </button>
          </div>
          <li
            className="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
            disabled={false}
          >
            <div
              className="filter-list-item"
            >
              <p
                className="MuiTypography-root MuiTypography-body1 filter-list-item__filter-name css-ahj2mt-MuiTypography-root"
              >
                Trainee
              </p>
              <button
                className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium filter-list-item__delete-button css-78trlr-MuiButtonBase-root-MuiIconButton-root"
                disabled={false}
                onBlur={[Function]}
                onContextMenu={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={0}
                type="button"
              >
                +
              </button>
            </div>
          </li>
          <li
            className="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
            disabled={false}
          >
            <div
              className="filter-list-item"
            >
              <p
                className="MuiTypography-root MuiTypography-body1 filter-list-item__filter-name css-ahj2mt-MuiTypography-root"
              >
                Junior
              </p>
              <button
                className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium filter-list-item__delete-button css-78trlr-MuiButtonBase-root-MuiIconButton-root"
                disabled={false}
                onBlur={[Function]}
                onContextMenu={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={0}
                type="button"
              >
                +
              </button>
            </div>
          </li>
        </ul>
      </div>
      <div
        className="filter-list"
      >
        <ul
          className="MuiList-root MuiList-padding css-h4y409-MuiList-root"
          data-testid="filter-list-test"
        >
          <div
            className="filter-list__name"
          >
            <p
              className="MuiTypography-root MuiTypography-body1 title css-ahj2mt-MuiTypography-root"
            >
              AUDITOR
            </p>
            <button
              className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium plus-button css-78trlr-MuiButtonBase-root-MuiIconButton-root"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              +
            </button>
          </div>
          <li
            className="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
            disabled={false}
          >
            <div
              className="filter-list-item"
            >
              <p
                className="MuiTypography-root MuiTypography-body1 filter-list-item__filter-name css-ahj2mt-MuiTypography-root"
              >
                Alex Vagner
              </p>
              <button
                className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium filter-list-item__delete-button css-78trlr-MuiButtonBase-root-MuiIconButton-root"
                disabled={false}
                onBlur={[Function]}
                onContextMenu={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={0}
                type="button"
              >
                +
              </button>
            </div>
          </li>
          <li
            className="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
            disabled={false}
          >
            <div
              className="filter-list-item"
            >
              <p
                className="MuiTypography-root MuiTypography-body1 filter-list-item__filter-name css-ahj2mt-MuiTypography-root"
              >
                Tom Cruise
              </p>
              <button
                className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium filter-list-item__delete-button css-78trlr-MuiButtonBase-root-MuiIconButton-root"
                disabled={false}
                onBlur={[Function]}
                onContextMenu={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={0}
                type="button"
              >
                +
              </button>
            </div>
          </li>
        </ul>
      </div>
    </div>
  </div>
</div>
`;
